src/afl-fuzz-one.c

```diff
@@ -2376,15 +2337,16 @@ u8 fuzz_one_original(afl_state_t *afl) {
 
             } else {
 
+              /* insert mode */
+
               u32 clone_from, clone_to, clone_len;
 
               clone_len = choose_block_len(afl, new_len);
               clone_from = rand_below(afl, new_len - clone_len + 1);
+              clone_to = rand_below(afl, temp_len + 1);
```

Comment:
the +1 here means at end of the data, IMHO this was missing. WDYT?

Reply:
I tried to understand it, but wouldn't that mean if `clone_from` and `clone_to` both result in max, we will write out of bounds?

Reply:
no because the tail copy is "temp_len - clone_to" so if the rand results in temp_len, then 0 bytes are copied. there iis already a +1 extra on the realloc which is unnecessary. at least that is how I read the code :)